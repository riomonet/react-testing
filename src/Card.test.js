import React from 'react'
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";


//smoke test
test("it renders without crashing", () => {
    render(<Card />)
})

//snapshot test

test('it mathces snapshot', () => {
    const {asFragment} = render(<Card />);
    expect(asFragment()).toMatchSnapshot()
    
})

it('matches snapshot with props', () => {
    const {asFragment} = render(<Card caption="A nice pic" currNum={1} totalNum={3} />);
    expect(asFragment()).toMatchSnapshot()

})
