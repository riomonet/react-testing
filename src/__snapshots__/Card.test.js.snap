// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it mathces snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with props 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      A nice pic
    </h4>
    <img
      alt="A nice pic"
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
